clarify unmentioned constraint port.name equalness (#11680)
istio/istio.io: content/en/docs/reference/glossary/namespace-sameness.md

Within a multicluster mesh, [namespace sameness](https://github.com/kubernetes/community/blob/master/sig-multicluster/namespace-sameness-position-statement.md)
applies and all namespaces with a given name are considered to be the same namespace. If multiple clusters contain a

> `Service` with the same namespaced name, they will be recognized as a single combined service. By default, traffic is

Reviewer: Do you know what other pages you looked at and didn't find this information?

This is the glossary definition for namespace sameness.  What about "same namespaced name and port name"?

Author: I mean `Service.spec.ports[_].name`, as pined a link https://github.com/istio/istio/blob/93e4ec424b52da807228bd7784c59d646ec73733/pilot/pkg/xds/endpoint_builder.go#L240-L242

if the multicluster `Service` have same port, but each with a different name, they are no longer `same`, only the service in primary cluster affect.

```yaml
apiVersion: v1
kind: Service
metadata:
  name: nginx-service
spec:
  selector:
    app.kubernetes.io/name: proxy
  ports:
  - name: name-of-service-port       # <-- I mean this, service.spec.ports[_].name
    protocol: TCP
    port: 80
    targetPort: http-web-svc
```

Reviewer: I understand the situation, but I'm just trying to clarify: "namespace sameness" is a thing that exists in Kubernetes, and what Istio is doing is saying "service and namespace sameness" is required.  Let me propose an edit for this.

> `Service` with the same namespaced name and the same port name, then they will be recognized as a single combined service. By default, traffic is

Reviewer 2: This seems a bit misleading. It seems to be saying that "service with same port name", which suggests a service has 1 port - it can have many. Additionally, ports not matching doesn't mean the whole _Service_ is not considered 'same'; rather, just the port is not considered.

Author: Thanks for reply.
Sorry, my words did misleading. Let me give a example:

```yaml
# the one in primary cluster
apiVersion: v1
kind: Service
metadata:
  name: nginx-service
spec:
  selector:
    app.kubernetes.io/name: proxy
  ports:
  - name: primary-cluster-80
    protocol: TCP
    port: 80
    targetPort: http-web-svc

# the one in a remote cluster
apiVersion: v1
kind: Service
metadata:
  name: nginx-service
spec:
  selector:
    app.kubernetes.io/name: proxy
  ports:
  - name: remote-cluster-80
    protocol: TCP
    port: 80
    targetPort: http-web-svc
  - name: remote-cluster-irrelevant-port
    protocol: TCP
    port: 8080
    targetPort: irrelevant-port
```

These two service, in same namespace, had same name, both had 80 port, the one in remote cluster, also had an extra port, that's OK, service can have many ports.

But, for the 80, these two service had different `port.name`, one is `primary-cluster-80`, the other is `remote-cluster-80`.
If so, the eds of cds `outbound|80||nginx-service.default.svc.cluster.local` only contains endpoints in primary cluster, include the eastwest gateway. Pods in remote make request to this service, will always goes to workloads in primary cluster.

I will try to choose other words and ask for some help around, to express this more clear.

Reviewer 2: Yeah I argee, the poit I was trying to make is the granularity is Port, not Service -- if one of the port names matched they would still be counted

> `Service` with the same namespaced name, in addition same port within the same service should make sure `port.name` literally equal, then they will be recognized as a single combined service. By default, traffic is

Reviewer: ```suggestion
`Service` with the same namespaced name and the same port name, then they will be recognized as a single combined service. By default, traffic is
```

Author: Thanks. My words didn't express well. It took me quit a long time to debug my cluster environment, so I though this PR can help others too. :smiling_face_with_three_hearts:

Within a multicluster mesh, [namespace sameness](https://github.com/kubernetes/community/blob/master/sig-multicluster/namespace-sameness-position-statement.md)
applies and all namespaces with a given name are considered to be the same namespace. If multiple clusters contain a
`Service` with the same namespaced name, they will be recognized as a single combined service. By default, traffic is
load-balanced across all clusters in the mesh for a given service.

> For one or many ports within the combined service, same port value should also have same port name.

Reviewer: ```suggestion
Ports that match on _number_ must also have the same port _name_ to be considered as a combined service port.```
